client/state/login/actions.js

```diff
@@ -278,7 +283,40 @@ export const createSocialUser = ( service, token, flowName ) => dispatch => {
 			error,
 		} );
 
-		return Promise.reject( wpcomError );
+		return Promise.reject( error );
+	} );
+};
+
+/**
+ * Attempt to connect the current account with a social service
+ *
+ * @param  {String}    service    The external social service name.
+ * @param  {String}    token      Authentication token provided by the external social service.
+ * @param  {String}    redirectTo Url to redirect the user to upon successful login
+ * @return {Function}             Action thunk to trigger the login process.
+ */
+export const connectSocialUser = ( service, token, redirectTo ) => dispatch => {
```

Comment:
I think new endpoints should live under the new data layer, like so:
https://github.com/Automattic/wp-calypso/blob/master/client/state/data-layer/wpcom/me/notification/settings/index.js

But maybe we should do that in a follow up PR.